research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,434 @@
+- An OS-level focus change occurs such that the window containing the component no longer has focus.
+  For example, the user switches to a different browser tab or switches to a different application.
+
+<p class="todo">
+  Land <a href="https://github.com/openui/open-ui/pull/243">#243</a> and link out to this canonical
+  definition of light dismissal for consistency (remove inline definition). Layout changes to the
+  popup or its anchor element are also ground for light dismissal, figure out how to surface this
+  special case.
+</p>
+
+When a `popup` element is shown and a light dismiss interaction occurs:
+
+1. If a keypress event for the `ESCAPE` key was fired, run [hiding a `popup` element
+   steps](#hiding-popup-steps), with the _candidate subject_ set to the top-most `popup` element in
+   the [popup stack](#pop-stack).
+2. If a focus change occurs, the user agent must set the _start node_ to the element which received
+   the focus event and run [Hiding currently-shown `popup` elements steps](#hiding-current-popups).
+3. Otherwise, the user agent must set the _start node_ to null and run [Hiding currently-shown
+   `popup` elements steps](#hiding-current-popups).
+
+<h3 id="hiding-current-popups">
+  Hiding currently-shown <code>popup</code> elements steps
+</h3>
+
+The **hiding currently-shown `popup` elements steps**, given an optional popup element
+_candidate subject_, an optional element _invoker_, and an optional element _anchor element_, and an
+optional element _start node_, are as follows.
+
+For each popup element _subject_ in the
+[node document](https://dom.spec.whatwg.org/#concept-node-document)'s [popup stack](#popup-stack),
+from top to bottom:
+
+1. If _candidate subject_ is null, skip to step 6.
+2. Otherwise, starting with the _candidate subject_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_
+   is an ancestor to the _candidate subject_. If this condition is met, then return.
+3. Otherwise, if the _invoker_ is not null, starting with the _invoker_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_
+   is an ancestor to the _invoker_. If this condition is met, then return.
+4. Otherwise, if the _anchor element_ is not null, starting with the _anchor element_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_
+   is an ancestor to the _anchor element_. If this condition is met, then return.
+5. If _start node_ is not null, starting with the _start node_, walk the
+   [flat tree](https://drafts.csswg.org/css-scoping/#flattening) to determine whether the _subject_
+   is the _start node_ or an ancestor to the _start node_. If either condition is met, then return.
+6. Otherwise, for this _subject_, run the
+   [hiding a `popup` element steps](#hiding-popup-steps), with _subject_ as _candidate subject_.
```

Comment:
Do you want to pass along _invoker_?

Reply:
Actually yeah, even if it's just to set that to null. Addressed in 5fc5041409d28c395e81cf30f039a7c0e269b7fe